worker/catacomb/catacomb.go

```diff
@@ -110,11 +110,11 @@ func Invoke(plan Plan) (err error) {
 	// This goroutine runs the work func and stops the catacomb with its error;
 	// and waits for for the listen goroutine and all added workers to complete
 	// before marking the catacomb's tomb Dead.
-	go func() {
-		defer catacomb.tomb.Done()
+	catacomb.tomb.Go(func() error {
 		defer catacomb.wg.Wait()
 		catacomb.Kill(runSafely(plan.Work))
-	}()
+		return nil
```

Comment:
another option would be to grab err with

err := runSafely(plan.Work)
catcomb.Kill(err)
return err

but I'm fine with this

Reply:
Returning nil demonstrates that the err from runSafely just needs to be passed to Kill() and is not needed to be returned. So I think it's more self documenting this way. The catacomb.Kill() ultimately sends the correctly post-processed err to the underlying tomb.